File: leetcode/trees/BinaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result=[] # initialize empty list
        queue=[root] # append root into queue
        print(queue,"queue")
        while len(queue)  > 0: # queue length > 0
            length = len(queue) # goes level by level
            level=[] # inner list
            for i in range(length): # range(0,length) starts with 0 and stops at given second element
              node = queue.pop(0)
              if node: 
                level.append(node.val) 
                if node.left : # if node is true and it has left
                    queue.append(node.left) 
                if node.right:
                    queue.append(node.right) 
            if len(level) > 0:
                result.append(level)
        return result
            

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
    if (!root) return [];
    let outer = []
    let queue = [root]
    while (queue.length > 0) {
        let inner = []
        let levelSize = queue.length;
        for (let i = 0; i < levelSize; i++) {
            let node = queue.shift()
            inner.push(node.val)
            if (node.left) queue.push(node.left)
            if (node.right) queue.push(node.right)
        }
        outer.push(inner)
    }
    return outer
};